### DISCUSSION TITLE
How to disable suggestor functionality completely

### DISCUSSION THREAD

**Discussion by kee98** [oracle/opengrok#4266] [General]
I am trying to disable building suggestor due to following reason:
1. I have more than 500 large projects and I have noticed indexer remains busy most of time building suggestor and newly added projects are not picked for indexing soon.

To disable this, I have tried several options available on internet or in github documentation but I seems to be missing something and always suggestor gets prompt on UI as well as on container logs, I see logs like "INFO: Rebuilding the following suggesters: ,,, "

I have tried following options:

1. added docker environment variable: OPENGROK_SUGGESTTER_ENABLED: false
2. Removed suggestor section from /etc/configuration.xml file.
3. Tried adding false to suggestorConfig in read-only.xml (please note I have added only sugegstory config to read only, no other content):

```
-bash-4.2$ cat /opengrok/etc/read-only.xml 
  <void property="suggesterConfig">
     <void property="enabled">
       <boolean>false</boolean>
     </void>
  </void>
```

but still that seems to not work. Can anyone please help if I am missing something basic.

**Comment by vladak** [comment]
This is certainly not a valid configuration and the indexer should fail on it. It needs the usual Java object serialization XML tags, see https://github.com/oracle/opengrok/wiki/Read-only-configuration#generating-the-read-only-configuration for an example.

**Reply by kee98** [reply]
thanks this worked.
